client/host_queue.go

```diff
@@ -456,3 +450,23 @@ func (q *queue) Close() {
 	close(q.drainIn)
 	q.Unlock()
 }
+
+// errors
+
+type errQueueNotOpen string
```

Comment:
Implementing them as types is the [recommended Go approach](https://blog.golang.org/error-handling-and-go) to be able to check for specific errors, which we do in tests. Are we okay dropping that to make the change?